code/hsec-tools/src/Security/OSV.hs

```diff
@@ -0,0 +1,223 @@
+-- | This module contains the OSV datatype and its ToJSON instance.
+-- The module was initialized with http://json-to-haskell.chrispenner.ca/
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Security.OSV where
+
+import Data.Aeson (ToJSON(..), FromJSON(..), Value(..), (.:), (.=), object)
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Text (Text)
+
+data Affected = Affected
+  { affectedRanges :: [Ranges]
+  , affectedPackage :: Package
+  , affectedEcosystemSpecific :: EcosystemSpecific
+  , affectedDatabaseSpecific :: DatabaseSpecific
```

Comment:
There is also the `versions` field.  I'm OK if we exclude it and use only `ranges` within our tooling.  I would eventually like to extract the osv type to its own package so it can be generally useful for OSV processing.  For that we would need the fields that we (*security-advisories* as producer) don't use.  It's OK if we don't do that now - but maybe a comment is a good idea?

Reply:
I think we should include `versions`, there are packages not using SemVer in Hackage and [recommendation](https://ossf.github.io/osv-schema/#affectedrangestype-field) is to populate the `versions` field whenever `ranges.type` is not "SEMVER".

Reply:
Also, this seems to miss `severity` field

https://ossf.github.io/osv-schema/#affectedseverity-field